google-cloud-pubsub/lib/google/cloud/pubsub/subscriber/inventory.rb

```diff
@@ -22,30 +22,40 @@ class Subscriber
         ##
         # @private
         class Inventory
+          InventoryItem = Struct.new :bytesize, :pulled_at do
+            def self.from rec_msg
+              new rec_msg.to_proto.bytesize, Time.now
+            end
+          end
+
           include MonitorMixin
 
-          attr_reader :stream, :limit
+          attr_reader :stream, :limit, :bytesize, :extension
 
-          def initialize stream, limit
+          def initialize stream, limit:, bytesize:, extension:
             @stream = stream
             @limit = limit
-            @_ack_ids = []
+            @bytesize = bytesize
+            @extension = extension
+            @inventory = {}
             @wait_cond = new_cond
 
             super()
```

Comment:
Move `super()` above the creation of the new condition variable. (Otherwise it will crash on Ruby 2.7)

Reply:
PR #4504 was created to make this change, because this pattern is used in other places as well.

Reply:
This was fixed by #4504.